tsup ignores tsconfig.json `target` and uses node14

When I try to build an internal package in my monorepo using `tsup`, I see the following:
```
CLI Building entry: src/index.ts
CLI Using tsconfig: tsconfig.json
CLI tsup v6.3.0
CLI Target: node14
CJS Build start
✘ [ERROR] Top-level await is not available in the configured target environment ("node14")

    src/index.ts:11:4:
      11 │     await import("dotenv").then((dotenv)=>dotenv.config({
         ╵     ~~~~~

CJS Build failed
Error: Build failed with 1 error:
src/index.ts:11:4: ERROR: Top-level await is not available in the configured target environment ("node14")
    at failureErrorWithLog (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:1566:15)
    at /Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:1024:28
    at runOnEndCallbacks (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:1438:61)
    at buildResponseToResult (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:1022:7)
    at /Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:1134:14
    at responseCallbacks.<computed> (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:671:9)
    at handleIncomingPacket (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:726:9)
    at Socket.readFromStdout (/Users/jacob/code/selfbuilder/node_modules/esbuild/lib/main.js:647:7)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
```
    
 This does not make sense to me, since the target specified in tsconfig.json is `esnext`:
 ```
 ...
  "compilerOptions": {
    "module": "esnext",
    "target": "esnext",
  },
  ```
  
  Additionally, package.json specifies `"type": "module"`.
  
  Why does tsup ignore the target in tsconfig.json and use node14 instead?
  
  NOTE: It works (bundles successfully using esm) if I configure tsup directly, by adding the following to package.json:
  ```
  "tsup": {
    "target": "esnext",
    "format": ["esm"]
  },
  ```

--- a/src/index.ts
+++ b/src/index.ts
@@ -113,7 +113,7 @@ const normalizeOptions = async (
       }
     }
     if (!options.target) {
-      options.target = tsconfig.data?.compilerOptions?.target.toLowerCase()
+      options.target = tsconfig.data?.compilerOptions?.target?.toLowerCase()
     }
   } else if (options.tsconfig) {
     throw new PrettyError(`Cannot find tsconfig: ${options.tsconfig}`)

--- a/src/index.ts
+++ b/src/index.ts
@@ -113,7 +113,7 @@ const normalizeOptions = async (
       }
     }
     if (!options.target) {
-      options.target = tsconfig.data?.compilerOptions?.target
+      options.target = tsconfig.data?.compilerOptions?.target.toLowerCase()
     }
   } else if (options.tsconfig) {
     throw new PrettyError(`Cannot find tsconfig: ${options.tsconfig}`)

--- a/src/index.ts
+++ b/src/index.ts
@@ -53,7 +53,6 @@ const normalizeOptions = async (
     ...optionsOverride,
   }
   const options: Partial<NormalizedOptions> = {
-    target: 'node14',
     outDir: 'dist',
     ..._options,
     format:
@@ -113,10 +112,17 @@ const normalizeOptions = async (
         ...(options.dts.compilerOptions || {}),
       }
     }
+    if (!options.target) {
+      options.target = tsconfig.data?.compilerOptions?.target
+    }
   } else if (options.tsconfig) {
     throw new PrettyError(`Cannot find tsconfig: ${options.tsconfig}`)
   }
 
+  if (!options.target) {
+    options.target = 'node14'
+  }
+  
   return options as NormalizedOptions
 }
 

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -1102,3 +1102,44 @@ test('treeshake should work with hashbang', async () => {
     "
   `)
 })
+
+test('support target in tsconfig.json', async () => {
+  const { getFileContent } = await run(
+    getTestName(),
+    {
+      'input.ts': `await import('./foo')`,
+      'foo.ts': `export default 'foo'`,
+      'tsconfig.json': `{
+        "compilerOptions": {
+          "baseUrl":".",
+          "target": "esnext"
+        }
+      }`
+    }, 
+    {
+      flags: ['--format', 'esm', ],
+    }
+  )
+  expect(await getFileContent('dist/input.mjs')).contains('await import(')
+})
+
+test('override target in tsconfig.json', async () => {
+  await expect(
+    run(
+      getTestName(),
+      {
+        'input.ts': `await import('./foo')`,
+        'foo.ts': `export default 'foo'`,
+        'tsconfig.json': `{
+          "compilerOptions": {
+            "baseUrl":".",
+            "target": "esnext"
+          }
+        }`
+      }, 
+      {
+        flags: ['--format', 'esm', '--target', 'es2018' ],
+      }
+    )
+  ).rejects.toThrowError(`Top-level await is not available in the configured target environment ("es2018")`)
+})

